pkg/dependencies/factory.go

```diff
@@ -167,6 +168,19 @@ func (f *Factory) WithProxyConfiguration(proxyConfig map[string]string) *Factory
 	return f
 }
 
+func (f *Factory) WithProxyConfigurationFromEnv() *Factory {
+	if f.proxyConfiguration == nil {
```

Comment:
I think we could treat this as another dependency
for example: rename `WithProxyConfiguration` to `UseProxyConfiguration`
And have this new method named `WithProxyConfiguration`

Then any dependency that wants the proxy config, will call `WithProxyConfiguration` as part of their dependencies. And at the build time, if the config hasn't been set manually already (with `UseProxyConfiguration`) then we read it from the environment

Reply:
@g-gaston I *think* I did what you asked, let me know if it wasn't what you were imagining.

@vivek-koppuru re [this comment](https://github.com/aws/eks-anywhere/pull/2849#discussion_r937035140), I was concerned about the same thing.

I think that with the approach Guillermo outlines above the proxy config from the cluster config file will always take precedence over the env vars as long as `UseProxyConfiguration` is explicitly called as a dependency. I think this is what we want, right?